### PR TITLE
Remove page-based modals in favor of existing state-based modal logic

### LINKED ISSUE #1264: Migrate all URL-based modals (usePushModal) to the state-based system (dispatch(pushModal(...)))
Many modals currently use the `dispatch(pushModal())` / `actions.pushModal` system instead of the (presumably?) newer router-based `usePushModal` system. When touching these modals, we should migrate them to the new system.

### PR DESCRIPTION
**Pull request by j-f1**
Fixes #1264

### CODE REVIEW

**Comment by MatissJanis** [comment]
Can you explain this a bit more? I think I'm missing some context so I don't understand why this change is necessary and what problem it would solve.

**Comment by j-f1** [comment]
Right now, we have two parallel systems for displaying modals on top of the current page. One of them uses an array of modals stored in Redux state, then manually renders those modals in an overlay. The other one uses a linked list of modals stored as the current location, with the `locationPtr` state object pointing to the parent modal (or the root page object).

While the location-based modals offer some advantages (for example, they update the URL when invoked, and we can use React Router's matching logic to select which modals to display, it requires a few hacks:

1. The location object itself is stored in the location state in order to be restored when the modal is closed (rather than some sort of representation of the URL the location corresponds to)
2. Reloading the page transforms the modal into a full page, which has undesirable results in most cases because the modal was not designed to integrate well as a page. So we need to manually redirect back to a sensible page. Meanwhile, the Redux based modals just disappear when reloading (or we could theoretically save them) but the underlying page remains the same.
3. Similar to 2, the sidebar and any other components that use the current URL to change appearance behave incorrectly when a modal is opened because they begin to correspond to the modal, not the page they're visually adjacent to.
4. React Router is trying to move users to new APIs, and the model where we explicitly pass in a custom location object in order to render the page or modals underneath the frontmost modal would not work as is.
6. It's a bit confusing — where should new modals be added? Now there is only one place.
7. Technically, *any* page could be rendered as a modal. I think it's best to have to explicitly opt in components to being rendered as a modal, and explicitly update them to handle that.  Most components should probably be either always a page or always a modal.
8. As you can see in the PR, the current implementation overloads `<Page />` to also represent modals which is IMO a bad mismatch.

**Comment by MatissJanis** [comment]
Thanks for the explanation.

Yes, it makes sense to port everything over to a single system. Thanks for taking this up! I'll do a review once all the CI jobs pass.

**Comment by j-f1** [comment]
Fixed the CI!

File: packages/desktop-client/src/components/schedules/LinkSchedule.js [hunk 1]
```diff
@@ -25,18 +25,16 @@ export default function ScheduleLink() {
   let { schedules, statuses } = scheduleData;
 
   async function onSelect(scheduleId) {
-    let { state } = location;
-    let ids = state.transactionIds;
     if (ids && ids.length > 0) {
-      await send('transactions-batch-update', {
+      await send('tran;sactions-batch-update', {
```
Comment (MatissJanis) [inline packages/desktop-client/src/components/schedules/LinkSchedule.js:29]: oops

File: packages/desktop-client/src/components/schedules/index.js [hunk 2]
```diff
@@ -102,3 +101,5 @@ export default function Schedules() {
     </Page>
   );
 }
+
+export default connect(null, actions)(Schedules);
```
Comment (MatissJanis) [inline packages/desktop-client/src/components/schedules/index.js:105]: Didn't you merge a PR to remove all `connect`s just a couple of days ago?

**Review by MatissJanis — CHANGES_REQUESTED** [review]
Would appreciate a review from @trevdor too since he's our react-router expert.

File: packages/desktop-client/src/components/schedules/index.js (hunk 2 above)
Comment (j-f1) [reply packages/desktop-client/src/components/schedules/index.js]: Good catch! I think I originally wrote this code before doing that PR. Updated to `useActions`

**Comment by trevdor** [comment]
🦥 I'm slow to get to this, but aim to look at it later tonight.

**Review by trevdor — APPROVED** [review]
SO much easier to reason about when all modals follow a single paradigm. Thank you!

**Pull request merged**